### DISCUSSION TITLE
能解释一下这些 快捷键的意思吗

### DISCUSSION THREAD

**Discussion by Dingjinjin-chihiro** [nshen/learn-neovim-lua#123] [Q&A] [answered]
-- lsp 回调函数快捷键设置
pluginKeys.mapLSP = function(mapbuf)
  -- rename
  mapbuf("n", "<leader>rn", "<cmd>lua vim.lsp.buf.rename()<CR>", opt)
  -- code action
  mapbuf("n", "<leader>ca", "<cmd>lua vim.lsp.buf.code_action()<CR>", opt)
  -- go xx
  mapbuf("n", "gd", "<cmd>lua vim.lsp.buf.definition()<CR>", opt)
  mapbuf("n", "gh", "<cmd>lua vim.lsp.buf.hover()<CR>", opt)
  mapbuf("n", "gD", "<cmd>lua vim.lsp.buf.declaration()<CR>", opt)
  mapbuf("n", "gi", "<cmd>lua vim.lsp.buf.implementation()<CR>", opt)
  mapbuf("n", "gr", "<cmd>lua vim.lsp.buf.references()<CR>", opt)
  -- diagnostic
  mapbuf("n", "gp", "<cmd>lua vim.diagnostic.open_float()<CR>", opt)
  mapbuf("n", "gk", "<cmd>lua vim.diagnostic.goto_prev()<CR>", opt)
  mapbuf("n", "gj", "<cmd>lua vim.diagnostic.goto_next()<CR>", opt)
  mapbuf("n", "<leader>f", "<cmd>lua vim.lsp.buf.formatting()<CR>", opt)
  -- 没用到
  -- mapbuf('n', '<leader>q', '<cmd>lua vim.diagnostic.setloclist()<CR>', opt)
  -- mapbuf("n", "<C-k>", "<cmd>lua vim.lsp.buf.signature_help()<CR>", opt)
  -- mapbuf('n', '<space>wa', '<cmd>lua vim.lsp.buf.add_workspace_folder()<CR>', opt)
  -- mapbuf('n', '<space>wr', '<cmd>lua vim.lsp.buf.remove_workspace_folder()<CR>', opt)
  -- mapbuf('n', '<space>wl', '<cmd>lua print(vim.inspect(vim.lsp.buf.list_workspace_folders()))<CR>', opt)
  -- mapbuf('n', '<space>D', '<cmd>lua vim.lsp.buf.type_definition()<CR>', opt)
end

**Comment by allworldg** [comment]
在neovim 命令行输入 `:h vim.lsp.buf.signature_help()`
然后如法炮制

**Comment by nshen** [comment, marked as answer]
作为前端开发，我目前基本上只用到这几个

-- 重命名，在变量上敲击即可改名
mapbuf("n", "rn", "lua vim.lsp.buf.rename()", opt)
-- code action 在报错的地方比如eslint报错的地方敲击，会弹出建议修改的选项
mapbuf("n", "ca", "lua vim.lsp.buf.code_action()", opt)
-- 跳转到变量定义的地方
mapbuf("n", "gd", "lua vim.lsp.buf.definition()", opt)
-- 变量解释，类似vscode鼠标在变量上悬停的提示
mapbuf("n", "gh", "lua vim.lsp.buf.hover()", opt)
-- 跳转到引用该变量的的地方
mapbuf("n", "gr", "lua vim.lsp.buf.references()", opt)
-- 弹窗打开，比如变量报错信息太长显示不全的地方敲击即可弹窗中显示
mapbuf("n", "gp", "lua vim.diagnostic.open_float()", opt)
-- 跳转到上一个报错，下一个报错
mapbuf("n", "gk", "lua vim.diagnostic.goto_prev()", opt)
mapbuf("n", "gj", "lua vim.diagnostic.goto_next()", opt)

**Reply by Dingjinjin-chihiro** [reply]
谢谢哥了

**Comment by jinzhongjia** [comment]
上面都是一些lsp支持的操作，类型比如是变量的重命名，效果和jb家代码的重构（重命名）一样，剩下的就是比如定义查找，跳跃，code action是对该处代码可以进行的操作（gitsign也可以注册这里，然后操作git记录），上面作者说的很详细了，最好还是自己去看看各个插件的github页面，有一些作为没用的功能，对于我们自己可能很有效，比如目前作者用的还是fork的lspsaga,但是lspsaga原作者已经回来更新了，而且更新了很多内容
